reddit_service_activity/__init__.py

```diff
@@ -59,18 +59,45 @@ def record_activity(self, context, context_id, visitor_id):
         self.counter.record_activity(context.redis, context_id, visitor_id)
 
     def count_activity(self, context, context_id):
-        if not _ID_RE.match(context_id):
+        results = self.count_activity_multi(context, [context_id])
+        return results[context_id]
+
+    def count_activity_multi(self, context, context_ids):
+        if not all(_ID_RE.match(context_id) for context_id in context_ids):
             raise ActivityService.InvalidContextIDException
 
-        cache_key = "{context}/cached".format(context=context_id)
-        cached_result = context.redis.get(cache_key)
-        if cached_result:
-            return ActivityInfo.from_json(cached_result.decode())
-        else:
-            count = self.counter.count_activity(context.redis, context_id)
-            info = ActivityInfo.from_count(self.fuzz_threshold, count)
-            context.redis.setex(cache_key, _CACHE_TIME, info.to_json())
-            return info
+        activity = {}
+
+        cache_keys = [context_id + "/cached" for context_id in context_ids]
+        cached_info = context.redis.mget(cache_keys)
+        for context_id, cached_value in zip(context_ids, cached_info):
+            if cached_value is None:
+                continue
+            activity[context_id] = ActivityInfo.from_json(cached_value.decode())
+
+        missing_ids = [id_ for id_ in context_ids if id_ not in activity]
```

Comment:
🔕  A set difference might be faster here, but then you also have to cast to a set, and may be a negligible difference if we're assuming this will only get a few contexts at a time. I think I just convinced myself to not bother.